components/button/index.tsx

```diff
@@ -1,12 +1,21 @@
 import { ButtonHTMLAttributes } from 'react';
 
-export default function Button(
-  props: ButtonHTMLAttributes<HTMLButtonElement>
-): JSX.Element {
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  outline?: boolean;
+}
+
+export default function Button({
+  outline,
+  ...props
+}: ButtonProps): JSX.Element {
   return (
     <button
       {...props}
-      className="inline-flex justify-center py-2 px-11 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-gray-700 hover:bg-gray-800 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-700">
+      className={`inline-flex justify-center py-2 px-11 shadow-sm text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-700 ${
```

Comment:
I have a question regarding this: Will it be purged in Production? 

In [https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html](https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html), it is said that "Don't use string concatenation to create class names", but instead "Do dynamically select a complete class name"

Reply:
purgecss will not purge it. what it will purge is when using dynamic value for a part of className like

```js
const margin = 10

<button className=`m-${margin}`></button>
```

in the case above purgecss will not see that the `m-10` class is being used.

While the code in PR is doing as recommended by tailwind

<img width="833" alt="Screen Shot 2021-04-05 at 16 44 22" src="https://user-images.githubusercontent.com/22452017/113561037-2ea29100-962e-11eb-84a9-6390db8dba2f.png">